app/assets/javascripts/discourse/app/components/composer-editor.js

```diff
@@ -213,20 +215,55 @@ export default Component.extend(
       });
     },
 
+    _destroyTippyInstances() {
+      this.tippyInstances.forEach((instance) => {
+        instance.destroy();
+      });
+      this.tippyInstances = [];
+    },
+
     @on("didInsertElement")
     _composerEditorInit() {
       const $input = $(this.element.querySelector(".d-editor-input"));
 
       if (this.siteSettings.enable_mentions) {
         $input.autocomplete({
           template: findRawTemplate("user-selector-autocomplete"),
-          dataSource: (term) =>
-            userSearch({
+          dataSource: (term) => {
+            this._destroyTippyInstances();
```

Comment:
is that necesarry?

Reply:
It's for edge cases, but it's to be sure the element gets destroyed when it should. Here's an example without it:
 
![demo](https://github.com/discourse/discourse/assets/66427541/b7464068-0c71-43fe-baa8-2de125d74a69)

Reply:
Ok Apparently we don't have an on close autocomplete event :/

Reply:
Unfortunately, this doesn't handle all cases. There is another way to reproduce the same bug as on the video:
1. Start typing a mention
2. When the autocomplete popup appears, hover over a user so a tooltip with their status appears
3. Press <kbd>Esc</kbd>. The autocomplete popup will disappear, but the status tooltip will still be shown

I think we should add a close event, we can fire it in https://github.com/discourse/discourse/blob/6f8056572e973960580d7a039b07ba7be93366bb/app/assets/javascripts/discourse/app/lib/autocomplete.js#L149